### PR TITLE
Collection view

### PR DESCRIPTION
**Pull request by ThanyaniDev**
I have added an updated collection view screen
Simulator Screen Shot - iPhone 12 Pro Max - 2020-11-09 at 07 41 23
And I have also added functions that convert UTC to Local Date
Screenshot 2020-11-09 at 07 35 43

### CODE REVIEW

File: MarsMission/Controllers/LandingViewController.swift [hunk 1]
```diff
@@ -10,62 +10,60 @@ import UIKit
...
-	
+
 	func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-		let cell = collectionView.dequeueReusableCell(withReuseIdentifier: keys.Identifier.cellID, for: indexPath) as! weatherForecastViewCell
+		let cell = collectionView.dequeueReusableCell(withReuseIdentifier: keys.Identifier.weatherForecastCellId, for: indexPath) as! weatherForecastViewCell
 		cell.layer.cornerRadius = 10
-		cell.backgroundColor = .blue
-		cell.temp?.text = String(forecastWeatherData[indexPath.row].forecasts[indexPath.row].temp!)
+		cell.backgroundColor = .darkGray
+		cell.dateLabel.text = "weather"
```
Comment (Sashen943) [inline MarsMission/Controllers/LandingViewController.swift]: Remove these hard coded values

File: MarsMission/Controllers/LandingViewController.swift [hunk 2]
```diff
@@ -10,62 +10,60 @@ import UIKit
...
 
 extension LandingViewController: UICollectionViewDataSource, UICollectionViewDelegate, UICollectionViewDelegateFlowLayout {
 	func numberOfSections(in collectionView: UICollectionView) -> Int {
 		return 1
 	}
-	
+
 	func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-		return forecastWeatherData.count
+		return 6
```
Comment (Sashen943) [inline MarsMission/Controllers/LandingViewController.swift]: Should return the size of the array. Remove this hard coded 6

File: MarsMission/Models/WeatherData.swift [hunk 3]
```diff
@@ -8,19 +8,15 @@
 import Foundation
 
 struct WeatherData: Codable {
-	var forecasts: [forecastsData]
+	var forecasts: [forecast]
```
Comment (Sashen943) [inline MarsMission/Models/WeatherData.swift]: class Forecast should be named with a capital. Remember that all classes in swift are named with a capital

File: MarsMission/Models/WeatherData.swift [hunk 4]
```diff
@@ -8,19 +8,15 @@
...
 	var lastUpdated: String?
 	var weatherStation: String?
 }
 
-struct forecastsData: Codable {
+struct forecast: Codable {
```
Comment (Sashen943) [inline MarsMission/Models/WeatherData.swift]: Capitalise class name

File: MarsMission/ViewModels/WeatherDataViewModel.swift [hunk 5]
```diff
@@ -35,7 +36,10 @@ class WeatherDataViewModel {
 		DispatchQueue.main.async { [self] in
 			self.weatherData = weatherData
 			self.view.populateWeatherData(weatherData.lastUpdated!,
-										  weatherData.weatherStation!)
+										  weatherData.weatherStation!,
```
Comment (Sashen943) [inline MarsMission/ViewModels/WeatherDataViewModel.swift]: Why do you have this function if you're using a collection view? Make sure to understand how arrays work

File: MarsMission/Views/WeatherView.swift [hunk 6]
```diff
@@ -8,7 +8,6 @@
 import Foundation
 
 protocol WeatherView {
-	func populateWeatherData(_ lastUpdated: String,_ weatherStation: String)
+	func populateWeatherData(_ lastUpdated: String,_ weatherStation: String,_ date: String,_ temp: Double)
```
Comment (Sashen943) [inline MarsMission/Views/WeatherView.swift]: Why do you have this function if you're using a collection view? Please ensure that you understand how the UI Components that you are using works before you use them

File: MarsMission/Helper/UTCToDate.swift [hunk 7]
```diff
@@ -7,15 +7,12 @@
...
-		convertedUTC = formatter.string(from: currentDate!)
-		return convertedUTC
-	}
+func convertUTCDateToLocalDate(date:String) -> String {
+	let format = DateFormatter()
+	format.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+	let convertedDate =  format.date(from: date)
+	format.timeZone =  TimeZone.current
+	format.dateFormat = "MMM d, h:mm a"
+	let localDateString =  format.string(from: convertedDate!)
```
Comment (Sashen943) [inline MarsMission/Helper/UTCToDate.swift]: Remove forceful unwrapping here

File: MarsMission/Controllers/LandingViewController.swift [hunk 8]
```diff
@@ -10,62 +10,60 @@ import UIKit
...
 		viewModel.fetchWeatherData()
 		configureUI()
-		setUpCollectionView()
 		navigationItem.title = keys.navigationTitle.landingTitle
 	}
 }
 
 extension LandingViewController: WeatherView {
-	func setUpCollectionView() {
-		let nib = UINib(nibName: "weatherForecastViewCell", bundle: nil)
```
Comment (Sashen943) [inline MarsMission/Controllers/LandingViewController.swift:29]: Why is you nib named with a lower case?

**Review by Sashen943 — CHANGES_REQUESTED** [review]
Please book some time with me to understand how collection views and arrays work

**Comment by ThanyaniDev** [comment]
Screenshot 2020-11-11 at 05 50 18
Screenshot 2020-11-11 at 01 56 51
Added Project Structure Changes and added nil coalescing

**Comment by ThanyaniDev** [comment]
Screenshot 2020-11-11 at 07 48 27

Screenshot 2020-11-11 at 01 56 51

**Comment by ThanyaniDev** [comment]
Simulator Screen Shot - iPhone 12 Pro Max - 2020-11-12 at 03 50 53
Updated forecast View.
Simulator Screen Shot - iPhone 12 Pro Max - 2020-11-12 at 03 51 00
forecast detail view

**Review by Sashen943 — APPROVED** [review]

**Pull request merged**